src/NServiceBus.Persistence.AzureStorage/SagaPersisters/SecondaryIndices/SecondaryIndexPersister.cs

```diff
@@ -98,8 +98,8 @@ public SecondaryIndexPersister(Func<Type, Task<CloudTable>> getTableForSaga, Sca
 
                         try
                         {
-                            await table.ExecuteAsync(TableOperation.Delete(existingSecondaryIndexEntity)).ConfigureAwait(false);
-                            await table.ExecuteAsync(TableOperation.Insert(newSecondaryIndexEntity)).ConfigureAwait(false);
+                            //this single call replaces a pair of calls that did a Delete followed by an Insert
+                            await table.ExecuteAsync(TableOperation.InsertOrReplace(newSecondaryIndexEntity)).ConfigureAwait(false);
```

Comment:
I think we can't `InsertOrReplace`. Another worker could go through the very same sequence and already create the secondary index entry and work as it's properly stored. Using `InsertOrReplace` would replace the previously stored secondary index.